### ISSUE TITLE
Option to widen the command palette

### ISSUE THREAD

**Issue by Moghul** [microsoft/vscode#85374]
I have a number of files that are quite deep in the file structure and some of them share names at least partially. It's a common occurrence that I will search for a file, get several options, and be unable to discern which of the results are the exact one I'm looking for.

I want to be able to
a) Make the width of the command palette increase automatically to include the full path
b) Manually set a px width for the command palette

PS. If there is a way to do either of these already, I have been unable to find it, and would appreciate being told how.

**Comment by peternoges** [comment]
This would be great to have.  I thought this extension would lend a workaround but not sure if its css rules can target the "Quick Open" or not.. https://marketplace.visualstudio.com/items?itemName=iocave.customize-ui

**Comment by Moghul** [comment]
Thanks for the suggestion. I'll investigate when I have some time on my hands :)

**Comment by utilityboy** [comment]
@peternoges , @Moghul using the extension you posted, I had success with the following:

```
  "customizeUI.stylesheet": {
    ".quick-input-widget": "width:800px !important"
  }
```

YMMV!

**Comment by Moghul** [comment]
@utilityboy Thanks for the help! That actually did do something. It's off-center, but at least it's wide enough now for me to read the full path. It'll do nicely until the official feature is implemented.

**Comment by zzhjerry** [comment]
Based on [Moghul](https://github.com/Moghul)'s answer, for anyone who wants a full width Quick Open:

```json
{
  "customizeUI.stylesheet": {
    ".quick-input-widget": "width: 100% !important; left: 300px"
  }
}
```

**Comment by johnnyoshika** [comment]
@zzhjerry thank you, where would I specify that?

**Comment by shr1k4nt** [comment]
> @zzhjerry thank you, where would I specify that?

In settings.json file after you install this extension https://marketplace.visualstudio.com/items?itemName=iocave.customize-ui

**Comment by cesarviana** [comment]
> This is pretty hacky to get working on Ubuntu without opening up some permissions. Would be nice to be able to enable some custom CSS for UI tweaking.

@rlzaleski Did you find a way to get it working on ubuntu? What permissions we need to open? Thanks.

**Comment by johnnyoshika** [comment]
Ability to widen the command palette would be to extremely helpful in my monorepo

**Comment by asalvare3** [comment]
Although the [Customize UI](https://marketplace.visualstudio.com/items?itemName=iocave.customize-ui) extension linked in a previous comment no longer works at this time for VSCode v1.74+, I was able to use the [vscode-custom-css](https://marketplace.visualstudio.com/items?itemName=be5invis.vscode-custom-css) to achieve the same effect as @zzhjerry 's settings.json. Just create a css file that I'll call `widen-command-palette.css`, copy the following css into the file:
```
.quick-input-widget {
    width: 80% !important;
}
```
and then follow the "Getting Started" instructions for the vscode-custom-css extension to copy `widen-command-palette.css` into the `vscode_custom_css.imports` setting. I'd also advise disabling the monkey patch from the Customize UI extension, and then uninstalling Customize UI and Monkey Patch since they're broken, as I mentioned.

After restarting VSCode, the command palette should be wider, and this should work for v1.74+ as well. Hopefully this helps any users who're totally css-illiterate, like I was.

Just wanna say though: I don't see this workaround as a satisfying alternative to the devs natively providing an option to widen the command palette.

**Comment by johnnyoshika** [comment]
I'm surprised this option isn't built into vscode yet. Visual Studio (the .Net IDE) has had that feature for years.

**Comment by Moghul** [comment]
Thanks for the update on that @asalvare3, I had postponed upgrading because it broke this feature. Just goes to show how necessary it is.

**Comment by tony-thai** [comment]
Thanks all! Are there any updates from the VS code team on this front?

**Comment by cesarviana** [comment]
Just sharing my use case:

![image](https://user-images.githubusercontent.com/6049357/212888560-ac7ab4ec-62da-420f-96ac-06def954592c.png)

The path at the beginning is equal for all search results 

I would be happy if we could:
- widen the command palette, or
- instead of repeating the file path, we could only show the part that repeats and below the parts that are different, for example:

```
     long/path/that/makes/the/file/name/invisible
          -- folderAAAA/filename.extension
          -- folderBBBB/filename.extension
```

**Comment by johnnyoshika** [comment]
@cesarviana Mine is very similar. I'm running a rather large monorepo, so I often don't know which file I have selected. I have a 38" wide monitor, so it's a bummer that the command palette is so narrow.

**Comment by tony-thai** [comment]
@johnnyoshika @cesarviana same here. I might have to go down the road of just writing an extension to solve the problem. Just need a few hours of "free" time. Will keep you guys posted.

**Comment by bndkk** [comment]
These solutions wouldn't work for me, so I just used `find . -name ...` in the integrated terminal. Most of the time you can also just ctrl-click a file to open it.

**Comment by f1tzpatrick** [comment]
Thanks for the suggestions, tweaking the CSS has made this widget much easier for me to use. Another question/request for command pallete customization, can it be configured such that `quick-input-header` is displayed below `quick-input-list`? 
<img width="1738" alt="image" src="https://github.com/microsoft/vscode/assets/84053270/6e84c954-1ffb-41a5-a05e-377951eafaad">

**Comment by Moghul** [comment]
> CSS that I ended up using to keep the dialog centered and working at different screen widths:
> 
> ```css
> .quick-input-widget {
>     width: 80% !important;
>     max-width: 900px !important;
>     left: 50% !important;
>     transform: translate(-50%, 0) !important;
>     margin-left: unset !important;
> }
> ```
> 
> Probably don't need all of those important tags but I'm hurrying through it. You can swap out the max-width value for whatever you want the dialog size to be able to grow to

This doesn't work for me anymore. Does anyone have a working solution, and/or is anyone 'quick-pick'ing this?

**Comment by Moghul** [comment]
This extension https://github.com/drcika/apc-extension
With the following settings did the trick:

```
    "apc.stylesheet": {
        ".quick-input-widget": {
            "width": "80% !important",
            "left": "50% !important",
            "transform": "translate(-50%, 0) !important",
            "margin-left": "unset !important"
        }
    }
```

**Comment by julien-langlois** [comment]
As Moghul said, I'm also not very comfortable with the idea of giving privileges to install a module for injecting CSS/JS into VSCode. Overall, I agree with everyone that on wide screens, it's difficult to list and differentiate between files that have the same name (I work in a monorepo), since the command palette is too narrow. 

I think this should be natively integrated into a future VSCode release.

**Comment by Mjvolk3** [comment]
@Moghul  Were you able to get things to work? I was able to get things to work on VsCode Version: 1.81.0 without running admin. Not sure if something changed.

**Comment by Moghul** [comment]
No, the extensions that allowed UI changes are now also deprecated. This PR https://github.com/microsoft/vscode/pull/117862 is supposed to fix it, but it has been stagnant for some weeks.

**Comment by utilityboy** [comment]
It's very unfortunate there is no movement on this given https://github.com/microsoft/vscode/pull/117862 (thanks for that contribution btw).  For those who are ok with looking at somewhat minified CSS, you don't need an extension to get this into a workable state.

VSCode (at least on MacOS) load up CSS from package contents (I would expect this is similar on Windows given Electron).  The css with the quick picker is in the workbench file:

![image](https://github.com/microsoft/vscode/assets/130537/95b68d0b-75d8-4ac0-9cd4-652c126e418f)

When you open / edit this file you'll be prompted like such: 

![image](https://github.com/microsoft/vscode/assets/130537/95e6ae16-3ebd-400e-9717-40829488ea9b)

This should be fine to do given VS Code is opening itself (inception!)

The selector you're looking for is the same as I posted above so you'll see something like: 

`.quick-input-widget{position:absolute;width:600px;...'`

Do whatever you need to inside that block... for me, I don't care if it's centred or not... I just want to see my files so my change is simply: 

`.quick-input-widget{position:absolute;width:1000px !important;...`

Once you save this file, and **restart** VSCode, you'll get an error saying your installation is corrupt (since the checksums will fail as a result of this change).  You can accept this and choose not to show the message in the future, or you can  just have it show every time VSCode starts up and rejoice that you can see the entire path to the file you're working on.

![image](https://github.com/microsoft/vscode/assets/130537/c1fc78a3-29df-4d45-a606-66d9da3c0cd5)

Ugly, yes.  Functional, also yes. Standard caveats apply here folks... if you're not comfortable making changes like this, please don't.  Just sharing my experience, so YMMV, but I wish you luck.  Also note, this change will be overwritten with each VSCode update.

**Comment by CaptainVincent** [comment]
My initial inspiration for this work actually stemmed from this discussion thread, and I am immensely grateful for the discussions around `iocave.customize-ui`. These discussions allowed me to gradually approach my goal, and in the end, I created a small tool that aligns with my personal expectations. The visual style is as follows. 

<img width="915" alt="273497706-65ea2b4a-5d71-4f8c-93db-145e7ea32d54" src="https://github.com/microsoft/vscode/assets/12159865/ec5be47d-97d4-4c37-8648-af7fa16c4613">

I also shared my accomplishment on Reddit's [/r/vscode](https://www.reddit.com/r/vscode/comments/173jrx6/share_my_customize_quickinputwidget/), but I thought there might be friends here who love this style as well. So, I'm also sharing it with you all. Here is my [gist](https://gist.github.com/CaptainVincent/74a15cd9d9c450e961b867f69008ee6e) link. Hope you enjoy using it! ❤️

(In fact, I didn't specifically adjust the width, but you can still include your width requirements in the script.)

**Comment by maxnevans** [comment]
+1 to this feature. This would be nice to see out of the box. I'd expect some "adaptive width" option to enable. I write in C++ and there are very long paths to std headers (obviously I'm interested in the last path node only) that I have to hover with mouse for now to see which is quite annoying as this takes a lot of time.

**Comment by chall1123** [comment]
> Based on [Moghul](https://github.com/Moghul)'s answer, for anyone who wants a full width Quick Open:
> 
> ```json
> {
>   "customizeUI.stylesheet": {
>     ".quick-input-widget": "width: 100% !important; left: 300px"
>   }
> }
> ```

@zzhjerry i insert this cfg in setting.josn, but the quick open width  has not changed

**Comment by JannesMeyer** [comment]
@chall1123 I suppose you also have to [install this extension](https://marketplace.visualstudio.com/items?itemName=iocave.customize-ui)

**Comment by michaelschufi** [comment]
Addendum to https://github.com/microsoft/vscode/issues/85374#issuecomment-1680855232

For Windows, the file (at least the way I installed it) is here:
`C:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\workbench.desktop.main.css`

BTW: 
- It doesn't matter if e.g. Prettier formats the code. But it needs the `!important`.
- To center it, also update the margin-left to half the value of width.

My final css rule:
```css
.quick-input-widget {
  left: 50%;
  /* edited */
  margin-left: -450px !important;
  position: absolute;
  /* edited */
  width: 900px !important;
  z-index: 2550;
  -webkit-app-region: no-drag;
  border-radius: 6px;
}
```

**Comment by starball5** [comment]
Related on Stack Overflow: [How to increase the "search by file name" search bar size in VS Code?](https://stackoverflow.com/q/78734505/11107541)

Related: #143956

Related issue ticket about command center width: #174616 

Duplicated by #89072, #58040

**Comment by olhapi** [comment]
This would be very useful to have, I'm really struggling with this.

**Comment by studgeek** [comment]
Note, @utilityboy's workaround is working well for me (thanks!) without having to install an injecting extension. Just wanted to say it explicitly so others know.

**Comment by mutantsan** [comment]
Can we have a proper solution for this one? Making the VS Code more customizable is always cool.

**Comment by Moghul** [comment]
Since they haven't done it in 6 years and have ignored the 4 year-old PR, apparently we can not...

**Comment by olhapi** [comment]
Solution to that is to switch to Zed

**Comment by billdotdev** [comment]
Here's a script that works on mac without any other plugins.
You would probably need to check the paths of the contents on linux and windows.

```ts
import fs from "fs/promises";
import { createHash } from "crypto";
import path from "path";

// Base app path
const appRoot = "/Applications/Visual Studio Code.app/Contents/Resources/app"; // VS Code
// const appRoot = "/Applications/Cursor.app/Contents/Resources/app"; // Cursor
const cssPath = path.join(appRoot, "out/vs/workbench/custom.css");
const htmlPath = path.join(
  appRoot,
  "out/vs/code/electron-sandbox/workbench/workbench.html"
);
const productJsonPath = path.join(appRoot, "product.json");

// Step 1: Write custom.css
const cssContent = `
.quick-input-widget {
  left: 50% !important;
  margin-left: unset !important;
  max-width: 1400px !important;
  transform: translate(-50%, 0) !important;
  width: 40% !important;
  top: 20% !important;
}
`.trim();

await fs.writeFile(cssPath, cssContent, "utf8");

// Step 2: Modify workbench.html
let html = await fs.readFile(htmlPath, "utf8");

const cssTag = `<link rel="stylesheet" href="../../../workbench/workbench.desktop.main.css">`;
const cssHash = createHash("sha256")
  .update(cssContent)
  .digest("hex")
  .slice(0, 8);
const customTag = `<link rel="stylesheet" href="../../../workbench/custom.css?v=${cssHash}">`;

// Remove existing custom.css
html = html
  .split("\n")
  .filter((line) => !line.includes('href="../../../workbench/custom.css'))
  .join("\n");

// Insert new custom tag
html = html.replace(cssTag, `${cssTag}\n\t\t${customTag}`);

// Update html file
await fs.writeFile(htmlPath, html, "utf8");
console.log("Custom CSS updated with version:", cssHash);

// Step 3: Compute checksum of modified workbench.html
const contents = await fs.readFile(htmlPath);
const checksum = createHash("sha256")
  .update(contents)
  .digest("base64")
  .replace(/=+$/, "");

// Step 4: Update product.json
const productRaw = await fs.readFile(productJsonPath, "utf8");
const product = JSON.parse(productRaw);

if (
  product.checksums["vs/code/electron-sandbox/workbench/workbench.html"] !==
  checksum
) {
  product.checksums["vs/code/electron-sandbox/workbench/workbench.html"] =
    checksum;
  await fs.writeFile(productJsonPath, JSON.stringify(product, null, 2), "utf8");
  console.log("Checksum updated:", checksum);
} else {
  console.log("Checksum already up to date:", checksum);
}

```

**Comment by danwt** [comment]
please!

**Comment by wilhem-meignan** [comment]
+1 for this feature. I would expect to be able to resize the command palette horizontally and vertically with the mouse. Especially since it is already possible to detach it and move it around with the mouse.

**Comment by protoEvangelion** [comment]
Thanks @billdotdev for the suggestion! I've tried so many extensions & they work for a bit...until they don't. Nice thing about owning the script is that I can see exactly what is wrong & fix it. Case in point: looks like they changed a path on mac. So after modifying the html path to be `electron-browser` instead of `electron-sandbox` & reloading we are golden! Also added my own css which is a bit more dynamic:

```
.quick-input-widget {
  width: 80vw !important;
  left: 50% !important;
  transform: translateX(-50%) !important;
  margin-left: 0 !important;
  max-width: 1500px !important;
}
```

**Comment by microJ** [comment]
VSCode extension `Apc Customize UI` doesn't work for me what be installed and config settings.json and reload VSCode. Maybe need run `sudo chown -R $(whoami) $(which code)`, but it has risk.

Reference to [#issuecomment-1680855232](#issuecomment-1680855232), it's not work for my mac.

Here is my solution:

1.  Edit `/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html`
2. Add code to it
    <img width="500" alt="Image" src="https://github.com/user-attachments/assets/11cb3e91-9730-4358-a8cd-a666592c9e39" />
3. Save File & Reload VSCode window

**Comment by marl0rd** [comment]
wow. 5 years in the backlog for a simple ui feature...

**Comment by marl0rd** [comment]
We know this is "open source", and we seriously appreciate the team's massive effort. No pressure at all, but we were just chatting about how cool all the new AI stuff gets implemented right away, and it makes me wondering why a super basic improvement like this one has been hanging out in the backlog for over five years! 

We know MS have a ton on the AI plate, but prioritizing some of these foundational quality-of-life fixes would make so many people happy.

And yes, neovim has this fixed but VScode is VScode

**Comment by wadamczyk-insolutions** [comment]
<img width="1180" height="269" alt="Image" src="https://github.com/user-attachments/assets/6eaeb5a1-0121-4303-93ae-80e0a00a0d42" />
Five-plus years seems a little long for an issue that could be fixed in 5 minutes using Codex. It would only require a few lines of CSS/JS to be reviewed and merged.

Since the VS Code team is already making minor UI tweaks here and there, and this task isn't big, I really don't get why it is so hard to finally address this professionally. I've been manually updating the CSS after every single update to fix this using a snippet I wrote in 5 minutes (without AI) -- for 5 years and counting.

**Comment by mmcintyre123** [comment]
> In the latest update you ask for feedback on developing two new themes (https://code.visualstudio.com/updates/v1_109#_tell-us-what-you-think-about-our-new-themes-experimental) which are told to be a "work-in-progress". Great! Then maybe if you have time to develop two new themes, you could finally also fix this easily fixable bug which is taking 5+ years to be solved and would be a so much needed fix for developers working on large codebases? Since it is clear that at least somebody from vscode team _is_ already working on some visual improvements? Please?

@wadamczyk-insolutions Isn't VS Code open source? Have you tried submitting a pull request?  You can even do that with AI by telling your agent to use the `gh` command line tool to interact with GitHub. It looks like many people would appreciate this but haven't bothered to tweak it locally and don't have time to do so or to work on a pull request.  Since you've invested time already, we'd appreciate it if you would!

**Comment by wadamczyk-insolutions** [comment]
@mmcintyre123 
> > In the latest update you ask for feedback on developing two new themes (https://code.visualstudio.com/updates/v1_109#_tell-us-what-you-think-about-our-new-themes-experimental) which are told to be a "work-in-progress". Great! Then maybe if you have time to develop two new themes, you could finally also fix this easily fixable bug which is taking 5+ years to be solved and would be a so much needed fix for developers working on large codebases? Since it is clear that at least somebody from vscode team _is_ already working on some visual improvements? Please?
> 
> [@wadamczyk-insolutions](https://github.com/wadamczyk-insolutions) Isn't VS Code open source? Have you tried submitting a pull request? You can even do that with AI by telling your agent to use the `gh` command line tool to interact with GitHub. It looks like many people would appreciate this but haven't bothered to tweak it locally and don't have time to do so or to work on a pull request. Since you've invested time already, we'd appreciate it if you would!

Yeah, sure, but... why would I? The pull request that fixes the issue is *already there* (!) and was created by @dscherger on Mar 1, 2021 https://github.com/microsoft/vscode/pull/117862 , is *still* awaiting review, so why would I bother to create another one? Of course if I knew that'd fix the issue, I'd be more than happy to do so, as now I keep applying a manual fix of 10 CSS lines after every VS Code update to fix the issue already (and I wrote these 10 lines by myself 4 years ago without AI, based on @dscherger's original pull request)... :-)

@Moghul out of curiosity, what did you switch to?

**Comment by ipanasenko** [comment]
My attempt for a fix: https://github.com/microsoft/vscode/pull/296767
